superset/assets/src/chart/Chart.jsx

```diff
@@ -78,196 +56,141 @@ class Chart extends React.PureComponent {
-          viz_type: vizType,
-          start_offset: renderStart,
-          duration: Logger.getTimestamp() - renderStart,
-        });
-      } catch (e) {
-        console.warn(e); // eslint-disable-line
-        this.props.actions.chartRenderingFailed(e, chartId);
-      }
-    }
-  }
-
   render() {
-    const isLoading = this.props.chartStatus === 'loading' || !this.state.renderVis;
+    const {
+      width,
+      height,
+      actions,
+      chartAlert,
+      chartId,
+      chartStatus,
+      errorMessage,
+      onDismissRefreshOverlay,
+      onQuery,
+      queryResponse,
+      refreshOverlayVisible,
+      vizType,
+    } = this.props;
+
+    const isLoading = chartStatus === 'loading';
 
     // this allows <Loading /> to be positioned in the middle of the chart
-    const containerStyles = isLoading ? { height: this.height(), width: this.width() } : null;
+    const containerStyles = isLoading ? { height, width } : null;
+    const isFaded = refreshOverlayVisible && !errorMessage;
+    const renderStart = Logger.getTimestamp();
+
     return (
-      <div className={`chart-container ${isLoading ? 'is-loading' : ''}`} style={containerStyles}>
+      <div
+        className={`chart-container ${isLoading ? 'is-loading' : ''}`}
+        style={containerStyles}
+      >
         {this.renderTooltip()}
 
         {isLoading && <Loading size={50} />}
 
-        {this.props.chartAlert && (
+        {chartAlert && (
           <StackTraceMessage
-            message={this.props.chartAlert}
-            queryResponse={this.props.queryResponse}
+            message={chartAlert}
+            queryResponse={queryResponse}
           />
         )}
 
-        {!isLoading &&
-          !this.props.chartAlert &&
-          this.props.refreshOverlayVisible &&
-          !this.props.errorMessage &&
-          this.container && (
-            <RefreshChartOverlay
-              height={this.height()}
-              width={this.width()}
-              onQuery={this.props.onQuery}
-              onDismiss={this.props.onDismissRefreshOverlay}
-            />
-          )}
+        {!isLoading && !chartAlert && isFaded && (
+          <RefreshChartOverlay
+            width={width}
+            height={height}
+            onQuery={onQuery}
+            onDismiss={onDismissRefreshOverlay}
+          />
+        )}
 
-        {!isLoading &&
-          !this.props.chartAlert && (
-            <ChartBody
-              containerId={this.containerId}
-              vizType={this.props.vizType}
-              height={this.height}
-              width={this.width}
-              faded={
-                this.props.refreshOverlayVisible && !this.props.errorMessage
+        {!isLoading && !chartAlert && (
+          <SuperChart
+            className={`slice_container ${snakeCase(vizType)} ${isFaded ? ' faded' : ''}`}
+            chartType={vizType}
+            chartProps={this.prepareChartProps()}
+            onRenderSuccess={() => {
```

Comment:
I think ideally we could bind these as class functions so we don't create a new function each render (which triggers a child re-render). I think we might have to store `renderStart` as `this.renderStart` for the logging in that case.

Reply:
👍